src/Moryx/Configuration/PossibleValuesSerialization.cs

```diff
@@ -65,21 +67,59 @@ public override EntryPrototype[] Prototypes(Type memberType, ICustomAttributePro
                 list.Add(new EntryPrototype(value, prototype));
             }
             return list.ToArray();
-        }
-
+        }
+
         /// <see cref="T:Moryx.Serialization.ICustomSerialization"/>
+#if NET8_0
+        public override string[] PossibleValues(Type memberType, ICustomAttributeProvider attributeProvider)
+        {
+            var valuesAttribute = attributeProvider.GetCustomAttribute<PossibleValuesAttribute>();
+            var validationAttribute = attributeProvider.GetCustomAttribute<AllowedValuesAttribute>();
+            var validation = new EntryValidation();
+
+            if (validationAttribute != null)
+            {
+                object[] allowed = validationAttribute.Values;
+                validation.AllowedValues = Array.ConvertAll(allowed, item => item.ToString());
+            }
+
+            IEnumerable<string> values;
+            // Possible values for primitive collections only apply to members
+            if (valuesAttribute == null || IsPrimitiveCollection(memberType))
+            {

```

Comment:
You could simplify you changes to this method as follows:
```c# 
var validationAttribute = attributeProvider.GetCustomAttribute<AllowedValuesAttribute>();
var allowedValues = validationAttribute?.Values.Select(v => v.ToString()) ?? Enumerable.Empty<string>() ;
return allowedValues.Concat(base.PossibleValues(memberType, attributeProvider));
```
I think that would be all you needed. Please check, however, whether these lines would be better placed in the DefaultSerialization class, as the PossibleValues method there is called from here. I think it should be better placed there 😊
Of course you could then move the precompiler condition to the other class and leave this method fully unchanged